src/App.tsx

```diff
@@ -1,61 +1,177 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { ChangeEvent, useState } from 'react';
+
+export type Todo = {
+  id: number;
+  title: string;
+  completed: boolean;
+  userId: number;
+};
+
+export type User = {
+  id: number;
+  name: string;
+  username: string;
+  email: string;
+};
 
 export const App = () => {
+  const [todosList, setTodosList] = useState<(Todo & { user: User })[]>(() =>
+    todosFromServer.map(todo => {
+      const user = usersFromServer.find(u => u.id === todo.userId);
+
+      if (!user) {
+        alert('User not found');
+
+        return {
+          ...todo,
+          user: {
+            id: 0,
+            name: '',
+            username: '',
+            email: '',
+          },
+        };
+      }
+
+      return {
+        ...todo,
+        user,
+      };
+    }),
+  );
```

Comment:
The initialization of `todosList` assumes that `todosFromServer` and `usersFromServer` are always available and correctly formatted. Consider adding error handling or validation to ensure data integrity.